tests/phpunit/api/v3/JobProcessMailingTest.php

```diff
@@ -257,8 +259,15 @@ public function testMailNonProductionRun(): void {
     $this->assertEquals($result['error_message'], "Job has not been executed as it is a Staging (non-production) environment.");
 
     // Test with runInNonProductionEnvironment param.
-    $this->callAPISuccess('job', 'process_mailing', ['runInNonProductionEnvironment' => TRUE]);
-    $this->_mut->assertRecipients($this->getRecipients(1, 2));
+    try {
+      // Do not call with callAPIFailure because we want to test the exception here
+      civicrm_api3('job', 'process_mailing', ['runInNonProductionEnvironment' => TRUE]);
+      $this->fail('should not be reachable');
+    }
+    catch (\CRM_Core_Exception_PrematureExitException $e) {
+      $error = $e->errorData;
+      $this->assertMatchesRegularExpression('/Outbound mail has been disabled/', $error['title']);
+    }
```

Comment:
The more complicated one is that what's happening is that old api3 thing where it loses the original error and where it came from and reformats it and [rethrows it](https://github.com/civicrm/civicrm-core/blob/95fd84629a0193ef65e00e6be67bcffcbb54dbe3/api/api.php#L138) as an api exception (nowadays a CRM_Core_Exception). If I replace the whole catch block with this it works more better (and you can remove the addition to statusBounce):

```php
    catch (\CRM_Core_Exception $e) {
      $this->assertEquals('civiExit called', $e->getMessage());
      $statuses = CRM_Core_Session::singleton()->getStatus();
      $hasExpectedMessage = FALSE;
      foreach ($statuses as $status) {
        if (str_contains($status['text'], 'Outbound mail has been disabled')) {
          $hasExpectedMessage = TRUE;
          break;
        }
      }
      if (!$hasExpectedMessage) {
        $this->fail('Status messages should contain "Outbound mail has been disabled"');
      }
    }
```

(It seems a little verbose - there might be a way to simplify it just note that there actually are multiple status messages.)

Reply:
@demeritcowboy Thank you!! I cleaned up the code and managed to run locally as well.